res/schema.xml

```diff
@@ -481,4 +481,48 @@ METADATA
         );
     </sql>
   </revision>  
+  <revision version="31" min_compatible="3">
+    <description>
+      Tables for Serato library feature
+    </description>
+    <sql>
+     CREATE TABLE IF NOT EXISTS serato_library (
```

Comment:
OK, I'll have a look at the way the banshee feature implements it. I based this PR on the Rekordbox implementation which was merged recently, so I figured it to be the least legacy-ridden one.

Reply:
Oh, I was not aware that the Recordbox implementation also uses a persistent table. 
Does this make sense? In case of Rythmbox and Traktor it does not. 
We may consider if using SQL makes sense to access the data. But I doubt, because we already pull everything into memory.

Reply:
I had a look at this. I though I could just replace the database object with a connection to a new in-memory SQLite DB (`:memory:`), but unfortunately that won't work with the `BaseExternalPlaylistModel`. I'll have to investigate, but it looks like the banshee feature has reimplemented parts of it in a custom class inheriting from `BaseSqlTableModel`. In any case this looks like it's quite a bit of work. I'll have to check when I find the time to do that.

Reply:
I had another idea to get around a major refactor by just creating `TEMP` tables in the constructor, but unfortunaly they are local to the connection and do not work with our `DbConnectionPool` even if I enable `?cache=shared` in `mixxxdb.cpp`. :cry:

Reply:
OK, maybe this was the reason to continue with the legacy for RecordBox. 
So it is fine as it is. Thank you for investigating.

Reply:
Would it be preferable to remove the schema change anyway? I could create a regular (persistant) table in the constructor and drop it in the destructor. Don't know if this is better. It would avoid a future schema change if we implement a better way later on.

Reply:
This sounds appealing.
Dropping the table is probably the same amount of work compared to empty the table. We just need to handle the case Mixxx was exited early.

Does it make sense to use this for all temporary tables?

Reply:
I don't know. I think for this feature it makes it easier switch to something else later and requires no schema change. But I wouldn't refactor and cause new schema changes for existing features unless we come up with a better solution (e.g. `TemporarySqlTableModel` that we can inherit from and that works with in-memory databases).

Reply:
By the way, the destructor of features is never called at the moment. I'll have to investigate.